### ISSUE TITLE
`yarn rw g scaffold` does not produce correct typing for `Int` `@id` schema fields

### ISSUE THREAD

**Issue by zygopleural** [redwoodjs/redwood#3871]
# Expected behaviour

`yarn rw g scaffold <model>` should generate a `./web/src/pages/Model/ModelPage/ModelPage.tsx` with props:
```typescript
type ModelPageProps = {
  id: number
}
```

# What currently happens

The props are created as:
```typescript
type ModelPageProps = {
  id: Int
}
```
and `yarn rw type-check` fails with:
```shell
src/pages/Model/ModelPage/ModelPage.tsx:4:7 - error TS2304: Cannot find name 'Int'.
```

# Steps to reproduce

### 1. Create new app
`yarn create redwood-app --typescript test`

### 2. Add model to `schema.prisma`
```prisma
model Example {
  id        Int    @id @default(autoincrement())
  something String
}
```

### 3. Migrate
`yarn rw prisma migrate dev`

### 4. Scaffold
`yarn rw g scaffold example`

### 5. Check
`yarn rw type-check`

should produce `src/pages/Example/ExamplePage/ExamplePage.tsx:4:7 - error TS2304: Cannot find name 'Int'.`

**Comment by aggmoulik** [comment]
Hey @jtoar Will try to fix it with the #3893

**Comment by orta** [comment]
The page props should be replaced via https://github.com/redwoodjs/redwood/issues/3875 then it doesn't need to be re-declared that file at all 👍🏻

**Comment by thedavidprice** [comment]
cc @callingmedic911 

This is a strange bug and I can confirm it's (unfortunately) alive an well as of today.

What's confusing is that `EditPostPage.tsx` uses the same template syntax as `PostPage.tsx`. However, `EditPostPage.tsx` has the correct typing `id: number` while `PostPage.tsx` ends up with `id: Int`. There must be something fishy here but it's not obvious at first glance:
https://github.com/redwoodjs/redwood/blob/6ea764e533f9a5c2282a2958f158960a4e5a2c77/packages/cli/src/commands/generate/helpers.js#L233-L244

@aggmoulik this issue is showing up in the test project, which I'd like to use for CI. Do you have time to work on this? If so:
- a fix keeping all things the same is very welcome
- if you'd like to try and improve via Orta's suggestion in #3875 that would be even better

@callingmedic911 do you see anything obvious to fix with `mapPrismaScalarToPagePropTsType`?

**Comment by thedavidprice** [comment]
AH, scratch all that. Looks like Danny resolved it over here: https://github.com/redwoodjs/redwood/pull/4190

Problem is that I was looking at a Test Project generated with previous code (with the TS error) but comparing it with the latest `main` generator code. No wonder I couldn't find the bug... 😆

Anyway, closing this as resolved by #4190 

I do think #3875 is better, but it's definitely low priority overall and can wait.
